@here/harp-mapview/lib/MapViewTaskScheduler.ts

```diff
@@ -17,7 +17,7 @@ export class MapViewTaskScheduler extends THREE.EventDispatcher {
     private readonly m_taskQueue: TaskQueue;
     private m_throttlingEnabled: boolean = false;
 
-    constructor(private m_maxFps: number = DEFAULT_MAX_FPS) {
+    constructor(private m_maxFps: number = DEFAULT_MAX_FPS, private readonly m_mapView: MapView) {
```

Comment:
We should not give access to the whole MapView to other classes.
Also, you had to check the MapView state in several places in processPending(), it's easy to miss adding the check if that code changes.
If the problem is that the tasks are executed after the MapView was disposed, I suggest to call from MapView.dispose() a new method MapViewTaskScheduler.dispose() that clears the task queue.

Reply:
There is no such interface on the MapViewTaskScheduler to clear the task queue, so I thought this was more appropriate. Perhaps instead of passing in the MapView, I can just pass a function to test if the mapview is still available? I will think about it.

Reply:
I added a method to clear the tasks.